### PR TITLE
test/e2e/karmadactl_test.go: test get command

### PR DESCRIPTION
**Pull request by mohamedawnallah**
**Description**

In this commit, we test `karmadactl get` command on existing resource, non-existing resource, and invalid resource type.

**What type of PR is this?**

/kind failing-test

**Which issue(s) this PR fixes**:

Part of #4544

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/karmadactl_test.go [hunk 1]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+		member1Client = framework.GetClusterClient(member1)
+	})
+
+	ginkgo.Context("Test karmadactl get for existing resource", func() {
+		var namespace, podName string
+		var pod *corev1.Pod
+
+		ginkgo.BeforeEach(func() {
+			namespace = fmt.Sprintf("karmadatest-%s", rand.String(6))
+			podName = fmt.Sprintf("test-pod-%s", rand.String(6))
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:629]:
```suggestion
			namespace = fmt.Sprintf("karmadatest-%s", rand.String(RandomStrLength))
			podName = podNamePrefix + rand.String(RandomStrLength)
```

File: test/e2e/karmadactl_test.go [hunk 2]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+			pod = helper.NewPod(namespace, podName)
+
+			// Create the namespace and pod in the member cluster.
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:637]:
```suggestion
			ns := helper.NewNamespace(namespace)
```

File: test/e2e/karmadactl_test.go [hunk 3]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+
+		ginkgo.BeforeEach(func() {
+			namespace = fmt.Sprintf("karmadatest-%s", rand.String(6))
+			podName = fmt.Sprintf("non-existent-pod-%s", rand.String(6))
+			// Create the namespace in the member cluster.
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:669]: ditto

File: test/e2e/karmadactl_test.go [hunk 4]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+			gomega.Expect(strings.Contains(err.Error(), fmt.Sprintf("pods \"%s\" not found", podName))).Should(gomega.BeTrue())
+		})
+	})
+
+	ginkgo.Context("Test karmadactl get for non-existing resource in non-existing namespace", func() {
+		var namespace, podName string
+
+		ginkgo.BeforeEach(func() {
+			namespace = fmt.Sprintf("karmadatest-%s", rand.String(6))
+			podName = fmt.Sprintf("non-existent-pod-%s", rand.String(6))
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:692]: ditto

File: test/e2e/karmadactl_test.go [hunk 5]
```diff
@@ -610,3 +610,108 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
...
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(err.Error(), fmt.Sprintf("namespaces \"%s\" not found", namespace))).Should(gomega.BeTrue())
+		})
+	})
+
+	ginkgo.Context("Test karmadactl get with invalid input", func() {
+		var namespace string
+
+		ginkgo.BeforeEach(func() {
+			namespace = fmt.Sprintf("karmadatest-%s", rand.String(6))
```
Comment (zhzhuang-zju) [inline test/e2e/karmadactl_test.go:707]: ditto

**Review by zhzhuang-zju — COMMENTED** [review]
Only a few issues with using public constants/methods, others LGTM

**Comment by mohamedawnallah** [comment]
Hey @zhzhuang-zju, The test cases finished and all passed after addressing all the feedback :)

**Comment by zhzhuang-zju** [comment]
/lgtm
cc @Vacant2333 @XiShanYongYe-Chang

**Comment by mohamedawnallah** [comment]
Please ping me once when it is time for merge this so I rebase :)

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/approve
Hi @mohamedawnallah, this pr also need to rebase the master branch. 
It can wait until the #5192 has been merged.

Other influential PRs are in a similar situation.

**Comment by mohamedawnallah** [comment]
Rebasing it now... Once the GitHub CI tests finish I will ask for review :)

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/hold cancel

**Comment by XiShanYongYe-Chang** [comment]
Hi @mohamedawnallah when you fix the comments, you can close it.

**Comment by mohamedawnallah** [comment]
@XiShanYongYe-Chang which comments?

**Comment by mohamedawnallah** [comment]
The CI tests finished and all passed. I'm gonna ask for review now, please let me if anything I'm missing before get merged :)

**Comment by XiShanYongYe-Chang** [comment]
Some comments made by others can be closed after the current modification. Some discussion comments can be closed after an agreement is reached with the commenter.

**Pull request merged**